core/spot_price_test.go

```diff
@@ -124,3 +136,71 @@ func Test_spotPrices_average(t *testing.T) {
 		})
 	}
 }
+
+func Test_fetch(t *testing.T) {
+	tests := []struct {
+		name             string
+		config           *spotPrices
+		product          string
+		duration         time.Duration
+		availabilityZone *string
+		instanceTypes    []*string
+		data             []*ec2.SpotPrice
+		err              error
+	}{
+		{
+			name: "error",
+			config: &spotPrices{
+				data: []*ec2.SpotPrice{},
+				conn: connections{
+					ec2: mock{
+						resp: &ec2.DescribeSpotPriceHistoryOutput{
+							SpotPriceHistory: []*ec2.SpotPrice{},
+						},
+						er: errors.New("Error!"),
+					},
+				},
+			},
+			data: []*ec2.SpotPrice{},
+			err:  errors.New("Error!"),
+		},
+		{
+			name: "ok",
+			config: &spotPrices{
+				data: []*ec2.SpotPrice{},
+				conn: connections{
+					ec2: mock{
+						resp: &ec2.DescribeSpotPriceHistoryOutput{
+							SpotPriceHistory: []*ec2.SpotPrice{
+								{SpotPrice: aws.String("1")},
+							},
+						},
+					},
+				},
+			},
+			data: []*ec2.SpotPrice{
+				{SpotPrice: aws.String("1")},
+			},
+			err: errors.New(""),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.config.fetch(tc.product, tc.duration, tc.availabilityZone, tc.instanceTypes)
+			if len(tc.data) != len(tc.config.data) {
+				t.Errorf("Price data actual: %v\nexpected: %v", tc.config.data, tc.data)
+			}
+			if len(tc.data) > 0 {
+				str1 := *tc.data[0].SpotPrice
+				str2 := *tc.config.data[0].SpotPrice
+				if str1 != str2 {
+					t.Errorf("Price actual: %s, expected: %s", str2, str1)
+				}
+			}
+			if err != nil && err.Error() != tc.err.Error() {
```

Comment:
It looks like there is a logical problem here.
If the method returns you nil, but you expected an error, then it would fail while it should, or am I missing something? Similar comment seem to apply to other PR - included the one merged earlier this week.